### DISCUSSION TITLE
How to setup a development environment

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse/lemminx-maven#343] [Q&A]
lemminx-maven shows [some hard to track issues](https://github.com/eclipse/lemminx-maven/issues/created_by/laeubi) I therefore like to setup a dev environment where I can attach a debugger once a problem occurs.

I therefore like to know:

1. How to setup an IDE to develop  lemminx(-maven) code
2. How to build a local distribution
3. How to configure a standard Eclipse-IDE to use that distribution
4. How to configure it that the server opens a debug port

**Comment by HannesWell** [comment]
Maybe the `HOWTO-DEV` m2e.lemminx helps a bit in this regard, at least in combination with m2e:
https://github.com/eclipse-m2e/m2e-core/blob/master/org.eclipse.m2e.editor.lemminx/HOWTO-DEV.md

**Comment by mickaelistria** [comment]
LemMinX-Maven is a trivial Maven projects that can be imported in any IDE with standard import steps for Maven projects. There is not much to document about the setup and the goal is to keep it as it for as long as possible.
The m2e intergation is indeed much trickier and https://github.com/eclipse-m2e/m2e-core/blob/master/org.eclipse.m2e.editor.lemminx/HOWTO-DEV.md is the current document to serve as help.
